### DISCUSSION TITLE
Posts published through the API are missing in search

### DISCUSSION THREAD

**Discussion by albattran** [bluesky-social/atproto#1872] [Bluesky Lexicons]
Hey everyone,
  I am using the HTTP API to post to BlueSky but the posts are not showing up when I search on bsky.app/search
  Is there any special API calls I need to make, or a field/parameter I need to set to have the posts appear in the search results?

Thanks a bunch!

**Comment by devinivy** [comment]
You shouldn't need to do anything special to start getting indexed.  Would you mind sharing your did or handle to help us take a look?

**Reply by connexinet** [reply]
Thanks for the quick reply Devin!
did=did:plc:svwrwmdybtcrm3wrgpo4mwxm

This was posted through bsky.app:
https://bsky.app/profile/feditst.bsky.social/post/3kenk4plnok27

This was posted through the API:
https://bsky.app/profile/feditst.bsky.social/post/3kenk5n5bh22y

**Reply by devinivy** [reply]
Alright, think I got to the bottom of this!  In short, it's an issue on our side that will be fixed before long.

What's happening is that the indexer for our search service didn't recognize the datetime format used in the post record: `"createdAt": "2023-11-20T21:18:32Z"`.  It expected subsecond precision on the time part, usually milliseconds.  However, the datetime format you're using is totally valid.  We're rolling out a new search service before long that handles this correctly, and your records with that datetime format (including this one) should appear there once we switch over.  In the meantime if you're concerned about appearing in search I'd suggest adding millisecond precision to your datetimes, e.g. `2023-11-20T21:18:32.000Z`.

**Reply by connexinet** [reply]
That worked! Thanks so much

Also amazing resolution! I have been working with social networks for 14 years, I don't think I got such a fast resolution like this (Not even when I was part of the Twitter Dev Insiders team)
You made me love working with BlueSky even more :-)

Thanks again!
